spectrwm.c

```diff
@@ -5504,7 +5504,7 @@ stack_master(struct workspace *ws, struct swm_geometry *g, int rot, bool flip)
 		/* Window coordinates exclude frame. */
 
 		if (winno > 1 || !disable_border ||
-		    (bar_enabled && ws->bar_enabled)) {
+		    (bar_enabled && ws->bar_enabled && winno > 1)) {
```

Comment:
Why do you actually need to check `bar_enabled` if the border is removed when there is only a single window regarless of whether the bar is enable or not?

Reply:
@mschilli87, below should make more sense I guess. 

```
if (winno > 1 || !disable_border)
        bordered = true;
else
        bordered = false;
```

Reply:
Yes, but why not just

```c
bordered = (winno > 1 || !disable_border);
```

?

Reply:
Yes, that is even better. Thank you. You can find the updated proposal at bket@479445a.